lib/src/typer.dart

```diff
@@ -25,23 +31,32 @@ class TyperAnimatedText extends AnimatedText {
           duration: speed * text.characters.length,
         );
 
-  Animation<int> _typingText;
+  Animation<double> _typingText;
 
   @override
   Duration get remaining => speed * (textCharacters.length - _typingText.value);
 
   @override
   void initAnimation(AnimationController controller) {
-    _typingText = StepTween(
-      begin: 0,
-      end: textCharacters.length,
+    _typingText = CurveTween(
+      curve: curve,
     ).animate(controller);
   }
 
   /// Widget showing partial text, up to [count] characters
   @override
   Widget animatedBuilder(BuildContext context, Widget child) {
-    final count = _typingText.value;
+    /// Output of CurveTween is in the range [0, 1] for majority of the curves.
+    /// It is converted to [0, textCharacters.length].
+    var count = (_typingText.value * textCharacters.length).round();
+
+    if (count < 0) {
```

Comment:
Could `count` ever be less than 0?  I don't think so.  I'd remove this.  If I am wrong, then please add a comment.

Reply:
Yes. `count` will indeed become less than zero. Look at the curve of `Curves.elasticInOut`. The output does go below zero.

![image](https://user-images.githubusercontent.com/35297280/103608096-8272e080-4f40-11eb-895d-826252db8943.png)

I was actually stuck at this part for a long time. An `assert` kept saying `count` is negative but I couldn't figure out where the problem was until I checked the documentation. This statement clips the value of `curve` when it goes below zero.

The value of `count` lies in `[0, 1]` for 30 curves. `Curves.elasticInOut`, `Curves.elasticOut`, `Curves.elasticIn`, and `Curves.easeOutBack` have otherwise.

Reply:
setting and upper and lower limit to `_typingText.value` using clamp should do the work

Reply:
Aw, snap! Thank you for teaching me something new here.  So, then a comment to _warn_ people of this may make sense to avoid future confusion.

Would this expression be too complicated to understand?
```dart
final count = (min(max(0, _typingText.value), 1.0) * textCharacters.length).round();
```

Reply:
`clamp` does the same thing just provide min and max values

Reply:
What I'm suggesting to do is

```dart
final count = _typingText.value.clamp(0, 1) * textCharacters.length
```

Reply:
@SirusCodes Better!  I learned another new thing.

Reply:
Nice ideas. I shall make the new commit with a cleaner code as soon as possible.